src/services/EcoBridge/OmniBridge/OmniBridge.ts

```diff
@@ -0,0 +1,800 @@
+        const response = await fetch(url)
+        if (response.ok) {
+          const json: TokenList = await response.json()
+          if (tokenListValidator(json)) {
+            return json.tokens.filter(token => token.chainId === from.chainId)
+          }
+        }
+        return []
+      }
+
+      const [homeData, foreignData, defaultTokens] = await Promise.all([
+        request<OmnibridgeSubgraphResponse>(homeEndpoint, homeTokensQuery),
+        request<OmnibridgeSubgraphResponse>(foreignEndpoint, foreignTokensQuery),
+        fetchDefaultTokens(),
+      ])
+
+      const allTokens = homeData.tokens.concat(foreignData.tokens, defaultTokens)
+
+      const uniqueTokens = () => {
+        const seen: { [address: string]: boolean } = {}
+        return allTokens.reverse().filter(token => {
+          const { address } = token
+          const lowerCaseAddress = address.toLowerCase()
+          const isDuplicate = Object.prototype.hasOwnProperty.call(seen, lowerCaseAddress)
+            ? false
+            : (seen[lowerCaseAddress] = true)
+          return isDuplicate
+        })
+      }
+
+      const tokens = uniqueTokens()
+
+      const tokenList: TokenList = {
+        name: 'OmniBridge',
+        timestamp: new Date().toISOString(),
+        version: VERSION,
+        tokens,
+      }
+
+      this.store.dispatch(this.actions.addTokenLists({ [this.bridgeId]: tokenList }))
+      this.store.dispatch(this.actions.setTokenListsStatus(SyncState.READY))
+    } catch (e) {
+      this.store.dispatch(this.actions.setTokenListsStatus(SyncState.FAILED))
+    }
+  }
+
+  public fetchStaticLists = async () => undefined
+
+  public getBridgingMetadata = async () => {
+    if (!this._activeProvider || !this._activeChainId || !this._staticProviders || !this._account) {
+      this.store.dispatch(this.actions.setBridgeDetailsStatus({ status: SyncState.FAILED }))
+      return
+    }
+
+    const {
+      address: fromTokenAddressUI,
+      chainId: fromChainId,
+      name: fromTokenName,
+      value: fromAmount,
+      decimals: fromTokenDecimals,
+      symbol: fromTokenSymbol,
+    } = this.store.getState().ecoBridge.ui.from
+
+    if (Number(fromAmount) === 0) {
+      this.store.dispatch(this.actions.setBridgeDetailsStatus({ status: SyncState.IDLE }))
+      return
+    }
+
+    const requestId = this.store.getState().ecoBridge[this.bridgeId as OmniBridgeList].lastMetadataCt
+
+    const helperRequestId = (requestId ?? 0) + 1
+
+    this.store.dispatch(this.actions.requestStarted({ id: helperRequestId }))
+
+    this.store.dispatch(this.actions.setBridgeDetailsStatus({ status: SyncState.LOADING }))
+
+    if (fromTokenAddressUI === Currency.getNative(this._homeChainId).symbol) {
+      this.store.dispatch(this.actions.setBridgeDetailsStatus({ status: SyncState.FAILED }))
+      return
+    }
+
+    const fromTokenAddress =
+      fromTokenAddressUI === Currency.getNative(this._foreignChainId).symbol ? ZERO_ADDRESS : fromTokenAddressUI
```

Comment:
Why are comparing address with the symbol? It doesn't look safe

Reply:
We pass native currency to store as `symbol` not as `address zero`